File: PredictTheNumber/c/ptn.c
// Predict the number solution in C for CodeEval.com by Steven A. Dunn

#include <stdio.h>
#include <stdlib.h>

long calculate(long);

int main (int argc, char* const argv[])
{
  FILE *fp = fopen(argv[1], "r");
  if (fp != NULL)
  {
    char line[256];
    while (fgets(line, sizeof(line), fp))
    {
      long n = atol(line);
      long x = calculate(n);
      int offset = 0;
      while (n > 0)
      {
        n -= x;
        offset++;
        x = calculate(n);
      }
      printf("%d\n", offset % 3);
    }
    fclose(fp);
  }
  return 0;
}

long calculate(long n)
{
  long x = 2;
  while (x <= n)
    x *= 2;
  return x/2;
}

File: PredictTheNumber/py2/ptn.py
# predict the number soln in py2 for code eval by steven a dunn

import sys

seq = [0, 1]

def calculate(n):
    x = 2
    while x <= n:
        x *= 2
    return x/2

f = open(sys.argv[1], 'r')
for line in f:
    n = int(line)
    x = calculate(n)
    offset = 0
    while n > 0:
        n -= x
        offset += 1
        x = calculate(n)
    print offset % 3
f.close()